drivers/platform/stm32/stm32_spi.c

```diff
@@ -45,6 +45,9 @@
 #include "stm32_spi.h"
 #include "no_os_delay.h"
 #include "no_os_alloc.h"
+#include "stm32_dma.h"
+#include "no_os_pwm.h"
+#include "stm32_pwm.h"
```

Comment:
is this pwm dependency really needed ?

Reply:
So the dependency we have is to control the chip select.
Since this is a DMA operation and cannot toggle chip select individually for each byte/word, we need a PWM module here

Reply:
As far as I know this functionality is not supported even for regular SPI transfers. Is there a use case for this? Why wouldn't you chain multiple byte/word transfers and handle the CS control in the transfer done interrupt?

Reply:
The goal of using a DMA is to read huge chunks of samples without interference of the core. The rate at which the read happens is at around 1MSPS, where in I would end up reading 2 bytes per conversion. Triggering interrupts at that high a rate just to control the state of the CS GPIO alone would be too much overhead on the core.

Hence, I am trying to control the CS via a PWM, where in I read n number of samples and get into the interrupt handler after reception of 400 samples and not after every byte